dev/tools/test/localization/gen_l10n_test.dart

```diff
@@ -308,6 +308,35 @@ void main() {
     expect(generator.header, '/// Sample header in a text file');
   });
 
+  test('sets templateArbFileName with more than one underscore correctly', () {
+    final Directory l10nDirectory = fs.currentDirectory.childDirectory('lib').childDirectory('l10n')
+      ..createSync(recursive: true);
+    l10nDirectory.childFile('app_localizations_en.arb')
```

Comment:
It would be usefulto verify that ap_localizations_en_CA is recognized as en_CA but app_localizations_en_CA_foo is not.

Reply:
I just tried `en_CA_foo` and it somehow parses the locale as `foo`. It seems that `Locale.tryParse` checks for the format of the locale identifier, so it would still be helpful to validate that the language is a valid one.